src/coreclr/jit/importer.cpp

```diff
@@ -7203,6 +7203,9 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     }
                 }
 
+                // Fold result, if possible.
+                op1 = gtFoldExpr(op1);
```

Comment:
This one is hit for `MATH_OP2`, so cases like `ADD`, `SUB`, `MUL`, `DIV`, `REM`, `AND`, `OR`, `XOR`